client/blocks/video-editor/index.jsx

```diff
@@ -0,0 +1,230 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { noop } from 'lodash';
+import { localize } from 'i18n-calypso';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import ProgressBar from 'components/progress-bar';
+import Notice from 'components/notice';
+import DetailPreviewVideo from 'post-editor/media-modal/detail/detail-preview-video';
+import VideoEditorButtons from './video-editor-buttons';
+import { updatePoster } from 'state/ui/editor/video-editor/actions';
+import {
+	getPosterUploadProgress,
+	getPosterUrl,
+	shouldCloseVideoEditorModal,
+	shouldShowVideoEditorError,
+} from 'state/selectors';
+
+/**
+ * Module variables
+ */
+let isSelectingFrame = false;
+
+class VideoEditor extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		media: PropTypes.object.isRequired,
+		onCancel: PropTypes.func,
+		onUpdatePoster: PropTypes.func,
+
+		// Connected props
+		posterUrl: PropTypes.string,
+		shouldCloseModal: PropTypes.bool,
+		shouldShowError: PropTypes.bool,
+		uploadProgress: PropTypes.number,
+	};
+
+	static defaultProps = {
+		onCancel: noop,
+		onUpdatePoster: noop,
+	};
+
+	state = {
+		error: false,
+		isLoading: true,
+		pauseVideo: false,
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! nextProps.shouldCloseModal && ! nextProps.shouldShowError ) {
+			return;
+		}
+
+		if ( nextProps.shouldShowError ) {
+			this.setState( {
+				error: true,
+				pauseVideo: false
+			} );
+
+			return;
+		}
+
+		this.props.onUpdatePoster( this.getVideoEditorProps( nextProps.posterUrl ) );
+	}
+
+	handleSelectFrame = () => {
+		const { isLoading } = this.state;
+
+		if ( isLoading ) {
+			this.setState( { error: true } );
+			return;
+		}
+
+		isSelectingFrame = true;
+
+		this.setState( {
+			error: false,
+			pauseVideo: true
+		} );
+	}
+
+	handlePause = ( currentTime ) => {
+		if ( ! isSelectingFrame ) {
+			return;
+		}
+
+		const { media } = this.props;
+		const guid = media && media.videopress_guid ? media.videopress_guid : null;
+
+		if ( guid ) {
+			this.props.updatePoster( guid, { atTime: Math.floor( currentTime ) } );
```

Comment:
If currentTime is in ms then`Math.floor( currentTime )` would be the same as `currentTime`.

If you were aiming to round to the nearest second, but keep the units in ms, that would be something like:
```
Math.round( currentTime / 1000 ) * 1000
```